controller/users.go

```diff
@@ -157,6 +192,95 @@ func (c *UsersController) getKeycloakProfileInformation(ctx context.Context, tok
 	return response, err
 }
 
+func (c *UsersController) updateUserInDB(id *uuid.UUID, ctx *app.UpdateUserAsServiceAccountUsersContext) error {
+
+	// We'll refactor the old Users update API to be redirected to Auth service in the near future.
+	// Hence, not spending time in refactoring that to consume this function.
+
+	returnResponse := application.Transactional(c.db, func(appl application.Application) error {
+		identity, err := appl.Identities().Load(ctx, *id)
+		if err != nil || identity == nil {
+			log.Error(ctx, map[string]interface{}{
+				"identity_id": id,
+			}, "id is unknown", *id)
+			jerrors, _ := jsonapi.ErrorToJSONAPIErrors(ctx, goa.ErrBadRequest(fmt.Sprintf("identity id %s is unknown", *id)))
```

Comment:
Well, if to be competently correct we should return BadRequest if ID is not in right format (not valid UUID) - it's handled above.
Internal Error if something wrong with executing the SQL request.
NotFound if no identity found with such ID.
:)

Reply:
Which means in this case we just can return jsonapi.ErrorToJSONAPIErrors(ctx, err). It will convert to 404 if no ID found and 500 if some other error.

Reply:
Got it, https://github.com/fabric8-services/fabric8-wit/pull/1637/commits/2c4369186b58b81b3b17700bf4b28f3f7768f2fd
Added a test too.